datasets/universal_morphologies/universal_morphologies.py

```diff
@@ -0,0 +1,498 @@
+    "Polarity": ["POS", "NEG"],
+    "Politeness": [
+        "INFM",
+        "FORM",
+        "ELEV",
+        "HUMB",
+        "POL",
+        "AVOID",
+        "LOW",
+        "HIGH",
+        "STELEV",
+        "STSUPR",
+        "LIT",
+        "FOREG",
+        "COL",
+    ],
+    "Possession": [
+        "ALN",
+        "NALN",
+        "PSS1S",
+        "PSS2S",
+        "PSS2SF",
+        "PSS2SM",
+        "PSS2SINFM",
+        "PSS2SFORM",
+        "PSS3S",
+        "PSS3SF",
+        "PSS3SM",
+        "PSS1D",
+        "PSS1DI",
+        "PSS1DE",
+        "PSS2D",
+        "PSS2DM",
+        "PSS2DF",
+        "PSS3D",
+        "PSS3DF",
+        "PSS3DM",
+        "PSS1P",
+        "PSS1PI",
+        "PSS1PE",
+        "PSS2P",
+        "PSS2PF",
+        "PSS2PM",
+        "PSS3PF",
+        "PSS3PM",
+    ],
+    "Switch_Reference": ["SS", "SSADV", "DS", "DSADV", "OR", "SIMMA", "SEQMA", "LOG"],
+    "Tense": ["PRS", "PST", "FUT", "IMMED", "HOD", "1DAY", "RCT", "RMT"],
+    "Valency": ["IMPRS", "INTR", "TR", "DITR", "REFL", "RECP", "CAUS", "APPL"],
+    "Voice": ["ACT", "MID", "PASS", "ANTIP", "DIR", "INV", "AGFOC", "PFOC", "LFOC", "BFOC", "ACFOC", "IFOC", "CFOC"],
+}
+
+_TAG_TO_CAT = dict([(tag, cat) for cat, tags in _CATEGORIES.items() for tag in tags])
+
+
+class UniversalMorphologies(datasets.GeneratorBasedBuilder):
+    """UniMorph: Universal Morphologies."""
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name=f"{language_iso}",
+            version=datasets.Version("1.0.0"),
+            description=f"Universal Morphologies for {language_iso}",
+        )
+        for language_iso in _LANGUAGES
+    ]
+
+    DEFAULT_CONFIG_NAME = "eng"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "lemma": datasets.Value("string"),
+                "forms": datasets.Sequence(
+                    dict(
+                        [("word", datasets.Value("string"))]
+                        + [
+                            (cat, datasets.Sequence(datasets.ClassLabel(names=tasks)))
+                            for cat, tasks in _CATEGORIES.items()
+                        ]
+                        + [("Other", datasets.Sequence(datasets.Value("string")))]  # for misspecified tags
+                    )
+                ),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,  # Here we define them above because they are different between the two configurations
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URLs)
```

Comment:
Could you download only the files related to the current config here ?
Currently it downloads everything.

You'll need to regenerate the dataset_infos.json after this change

Reply:
this will also simplify the dataset_infos.json file significantly (currently it's 3MB)